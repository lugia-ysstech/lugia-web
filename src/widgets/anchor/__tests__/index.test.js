/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import chai from 'chai';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { delay } from '@lugia/react-test-utils';
import AnchorDemo from '../demo';
import Anchor from '../anchor';

const { expect: exp } = chai;

Enzyme.configure({ adapter: new Adapter() });

describe('Anchor', () => {
  it('css', () => {
    const Target = <AnchorDemo />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });

  const getComponent = (target: any) => {
    return target
      .find('Anchor')
      .at(0)
      .instance();
  };

  it('Anchor: setScrollActiveLink ', () => {
    const target = mount(<Anchor />);
    const component = getComponent(target);

    component.setScrollActiveLink([]);
    const { activeLink: newActiveLink } = component.state;
    expect(newActiveLink).toBe('');

    component.setScrollActiveLink([
      { link: 'a', top: 20 },
      { link: 'b', top: -20 },
      { link: 'c', top: 0 },
    ]);
    const { activeLink } = component.state;
    expect(activeLink).toBe('a');
  });

  it('Anchor: handleLinkClick ', async () => {
    const target = mount(<Anchor />);
    const component = getComponent(target);
    const { isClick } = component;
    expect(isClick).toBe(false);

    component.handleLinkClick('', '#a');
    const { activeLink } = component.state;
    expect(component.isClick).toBe(true);
    expect(activeLink).toBe('#a');
    await delay(500);
    expect(component.isClick).toBe(false);
  });

  it('Anchor: getId ', () => {
    const target = mount(<Anchor />);
    const component = getComponent(target);

    expect(component.getId('#a')).toBe('a');
    expect(component.getId('##a')).toBe('a');
    expect(component.getId()).toBeUndefined();
  });

  it('Anchor: updateLinks ', () => {
    const target = mount(<Anchor />);
    const component = getComponent(target);
    component.updateLinks([1, 2]);
    const { links } = component;
    expect(links).toEqual([1, 2]);

    component.updateLinks([3, 4]);
    const { links: newLink } = component;
    expect(newLink).toEqual([3, 4]);
  });

  it('Anchor: getMaxTopLink ', () => {
    const target = mount(<Anchor />);
    const component = getComponent(target);
    const result = component.getMaxTopLink();
    expect(result).toBe('');

    const wrongMax = component.getMaxTopLink({});
    expect(wrongMax).toBe('');

    const max = component.getMaxTopLink([{ top: 1, link: '1' }, { top: 2, link: '2' }]);
    expect(max).toBe('2');
  });
});
