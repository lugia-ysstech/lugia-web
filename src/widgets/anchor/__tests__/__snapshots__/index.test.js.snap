// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor css 1`] = `
.c1 {
  font-size: 1.2rem;
  top: 0em;
}

.c2 {
  font-size: 1.2rem;
  position: relative;
  background-color: #fff;
  border-left: 0.08333333333333334em solid #cccccc;
  box-sizing: border-box;
  padding-bottom: 0.5em;
}

.c2 > div:first-child {
  padding-top: 0.5em;
}

.c5 {
  position: absolute;
  width: 0.6666666666666667em;
  height: 0.6666666666666667em;
  border-radius: 8px;
  left: -0.04166666666666667em;
  display: none;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c7 {
  position: absolute;
  width: 0.16666666666666669em;
  height: 2em;
  left: -0.04166666666666667em;
  display: none;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c3 {
  font-size: 1.2rem;
  padding-top: 1em;
  padding-left: 0.8333333333333334em;
  line-height: 1;
}

.c4 {
  display: block;
  color: #646A6E;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  color: #4d63ff;
}

.c4:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  padding: 20px;
  margin-left: 120px;
  width: 209px;
  height: 203px;
}

.c6 {
  padding: 20px;
  margin-left: 240px;
  width: 209px;
  height: 203px;
}

.c8 {
  padding: 20px;
  margin-left: 360px;
  width: 209px;
  height: 203px;
}

<div>
  <div
    className="c0"
  >
    <div
      style={null}
    >
      <div
        className="c1"
      >
        <div
          className="c2"
        >
          <div
            className="c3"
          >
            <a
              className="c4"
              onClick={[Function]}
            >
              anchor-link1
            </a>
          </div>
          <div
            className="c3"
          >
            <a
              className="c4"
              onClick={[Function]}
            >
              anchor-link2
            </a>
          </div>
          <div
            className="c3"
          >
            <a
              className="c4"
              onClick={[Function]}
            >
              anchor-link3
            </a>
            <div
              className="c3"
            >
              <a
                className="c4"
                onClick={[Function]}
              >
                anchor-link3.1
              </a>
            </div>
            <div
              className="c3"
            >
              <a
                className="c4"
                onClick={[Function]}
              >
                anchor-link3.2
              </a>
            </div>
          </div>
          <div
            className="c3"
          >
            <a
              className="c4"
              onClick={[Function]}
            >
              anchor-link4
            </a>
          </div>
          <div
            className="c5"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      style={null}
    >
      <div
        className="c1"
      >
        <div
          className="c2"
        >
          <div
            className="c3"
          >
            <a
              className="c4"
              href="#anchor-link1"
              onClick={[Function]}
            >
              anchor-link1
            </a>
          </div>
          <div
            className="c3"
          >
            <a
              className="c4"
              href="#anchor-link2"
              onClick={[Function]}
            >
              anchor-link2
            </a>
          </div>
          <div
            className="c3"
          >
            <a
              className="c4"
              href="#anchor-link3"
              onClick={[Function]}
            >
              anchor-link3
            </a>
            <div
              className="c3"
            >
              <a
                className="c4"
                href="#anchor-link3.1"
                onClick={[Function]}
              >
                anchor-link3.1
              </a>
            </div>
            <div
              className="c3"
            >
              <a
                className="c4"
                href="#anchor-link3.2"
                onClick={[Function]}
              >
                anchor-link3.2
              </a>
            </div>
          </div>
          <div
            className="c3"
          >
            <a
              className="c4"
              href="#anchor-link4"
              onClick={[Function]}
            >
              anchor-link4
            </a>
          </div>
          <div
            className="c7"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="c8"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="#anchor-link1"
          onClick={[Function]}
        >
          anchor-link1
        </a>
      </div>
      <div
        className="c3"
      >
        <a
          className="c4"
          href="#anchor-link2"
          onClick={[Function]}
        >
          anchor-link2
        </a>
      </div>
      <div
        className="c3"
      >
        <a
          className="c4"
          href="#anchor-link3"
          onClick={[Function]}
        >
          anchor-link3
        </a>
        <div
          className="c3"
        >
          <a
            className="c4"
            href="#anchor-link3.1"
            onClick={[Function]}
          >
            anchor-link3.1
          </a>
        </div>
        <div
          className="c3"
        >
          <a
            className="c4"
            href="#anchor-link3.2"
            onClick={[Function]}
          >
            anchor-link3.2
          </a>
        </div>
      </div>
      <div
        className="c3"
      >
        <a
          className="c4"
          href="#anchor-link4"
          onClick={[Function]}
        >
          anchor-link4
        </a>
      </div>
      <div
        className="c7"
      />
    </div>
  </div>
  <div
    id="anchor-link1"
    name="anchor-link1"
    style={
      Object {
        "marginTop": "100px",
      }
    }
  >
    anchor-link1
  </div>
  <div
    id="anchor-link2"
    name="anchor-link2"
    style={
      Object {
        "marginTop": "300px",
      }
    }
  >
    anchor-link2
  </div>
  <div
    id="anchor-link3"
    name="anchor-link3"
    style={
      Object {
        "marginTop": "300px",
      }
    }
  >
    anchor-link3
  </div>
  <div
    id="anchor-link3.1"
    name="anchor-link3.1"
    style={
      Object {
        "marginTop": "200px",
      }
    }
  >
    anchor-link3.1
  </div>
  <div
    id="anchor-link3.2"
    name="anchor-link3.2"
    style={
      Object {
        "marginTop": "200px",
      }
    }
  >
    anchor-link3.2
  </div>
  <div
    id="anchor-link4"
    name="anchor-link4"
    style={
      Object {
        "marginTop": "300px",
      }
    }
  >
    anchor-link4
  </div>
</div>
`;
